### PR TITLE
[Docs] Add Field Descriptions Meshery CRDs

### PR DESCRIPTION
**Pull request by mercenarysre**
Add Field Descriptions Meshery CRDs based on #9978, @leecalcote @vishalvivekm @sudhanshutech.

### CODE REVIEW

**Comment by leecalcote** [comment]
Oooh! Very good.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 1]
```diff
@@ -73,8 +74,10 @@ spec:
               endpoint:
                 properties:
                   external:
+                    description: The external endpoint address that applications can use to connect to the Broker
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:77]: ...external to the cluster within which the Broker is running.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 2]
```diff
@@ -73,8 +74,10 @@ spec:
...
                     type: string
                   internal:
+                    description: The internal endpoint address that applications can use to connect to the Broker
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:80]: Should be a locally addressable IP address or DNS hostname (e.g.`broker.busybox.svc.cluster-domain.example`).

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 3]
```diff
@@ -131,30 +134,38 @@ spec:
...
+                        description: URL of the custom Broker to connect to
                         type: string
                     type: object
                   native:
                     properties:
                       name:
+                        description: Name of the native Broker to connect to
                         type: string
                       namespace:
+                        description: Namespace where the native Broker is deployed
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:146]: Is there a default namespace that we would commonly expect to see? Can this be any namespace?

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 4]
```diff
@@ -131,30 +134,38 @@ spec:
                   custom:
                     properties:
                       url:
+                        description: URL of the custom Broker to connect to
                         type: string
                     type: object
                   native:
                     properties:
                       name:
+                        description: Name of the native Broker to connect to
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:143]: It'd be great to get more on the intended use of this name. Does this name get exposed in Meshery UI?

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 5]
```diff
@@ -131,30 +134,38 @@ spec:
...
                 type: integer
               watch-list:
                 type: object
                 properties:
                   data:
+                    description: Specifies the resource types and their associated whitelist and blacklist for watching
                     type: object
                     properties:
                       whitelist:
+                        description: Whitelist of resources to watch for changes
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:165]: A link to Meshery Docs on where the whitelist and blacklist terms are defined would be great.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 6]
```diff
@@ -131,30 +134,38 @@ spec:
...
                 type: string
               size:
+                description: Desired size of the Broker
                 format: int32
                 type: integer
               watch-list:
                 type: object
                 properties:
                   data:
+                    description: Specifies the resource types and their associated whitelist and blacklist for watching
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:161]: "...for watching... ?"

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 7]
```diff
@@ -36,6 +36,7 @@ spec:
             description: BrokerSpec defines the desired state of Broker
             properties:
               size:
+                description: Desired size of the Broker
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:39]: "size" in terms of number of instances or amount of RAM or CPU cores?

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 1 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 2 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 3 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: The namespace here is that of the Broker instance to be deployed.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 4 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: The intended name here is name of the Broker instance to be deployed.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 5 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 6 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: I have rectified the description accordingly.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 7 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: I have inputted a better description for the "size".

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 8]
```diff
@@ -36,6 +36,7 @@ spec:
             description: BrokerSpec defines the desired state of Broker
             properties:
               size:
+                description: Desired size of the Broker, will be an integer value denoting the number of Broker instances that should be in one cluster, currently it is adviced to have one Broker instance in a cluster but that can be scaled vertically up or down depending on load
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:39]: not vertically, horizontally.

Comment (leecalcote) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml:39]: Just don't mention direction and we're fine.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 9]
```diff
@@ -73,8 +74,10 @@ spec:
               endpoint:
                 properties:
                   external:
+                    description: The endpoint external to the cluster within which the Broker is running, which the Meshery Server will connect to
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:77]: The Broker isn't running "within" and endpoint, but is "exposed on".

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 10]
```diff
@@ -73,8 +74,10 @@ spec:
...
                     type: string
                   internal:
+                    description: The endpoint internal to the cluster within which the Broker is running, will be a locally addressable IP address or DNS hostname (e.g.`broker.busybox.svc.cluster-domain.example`, `10.96.168.204:4222`)
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:80]: Please clarify whether the port is required here or is separately defined.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 11]
```diff
@@ -131,30 +134,38 @@ spec:
                   custom:
                     properties:
                       url:
+                        description: URL of an already running external Broker instance
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:137]: What is at this URL? What's its purpose? Is this the admin interface on NATS or the URL/port for subscribing / unsubscribing to topics?

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 12]
```diff
@@ -131,30 +134,38 @@ spec:
...
                         type: string
                     type: object
                   native:
                     properties:
                       name:
+                        description: Name of the Broker instance to be deployed
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:143]: Is there a default name that is used, unless customized? Where does this name show up? Only in `kubectl get` or in Meshery UI, too?

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 13]
```diff
@@ -131,30 +134,38 @@ spec:
...
                         type: string
                       namespace:
+                        description: Namespace that the Broker instance to be deployed would run in
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:146]:
```suggestion
                        description: Namespace within which the Broker instance to be deployed.
```

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 14]
```diff
@@ -131,30 +134,38 @@ spec:
...
                         type: string
                     type: object
                 type: object
               version:  
+                description: The current version of the meshery application as from its release
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:151]: Please always capitalize Meshery, unless you're referring to a logical item like the `meshery` Namespace.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 15]
```diff
@@ -131,30 +134,38 @@ spec:
...
                 type: string
               size:
```
Comment (leecalcote) [inline install/kubernetes/helm/meshery-operator/crds/crds.yaml:153]: Perhaps, best to refer to this as replicas / replica count, which is a more precise description as opposed to "size".

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 8 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 9 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 13 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 14 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Alright, Noted.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 15 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml:170]: Done.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml [hunk 16]
```diff
@@ -131,30 +134,38 @@ spec:
...
                 type: integer
               watch-list:
+                description: A ConfigMap that sets the resources e.g. pods,deployments and event types e.g. ADDED, MODIFIED… that the MeshSync instance tracks or ignores in the Kubernetes cluster
                 type: object
                 properties:
                   data:
                     type: object
                     properties:
                       whitelist:
+                        description: 'Whitelist resources are resources that the MeshSync should track its changes, More info: https://docs.meshery.io/reference/meshery-operator-crds#meshsync-crd-properties'
```
[the remark this answers is not in the document]
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: You welcome.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 11 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: The URL here is of an already running instance of the Broker, which is an alternative to specifying the native configuration of the Broker instance.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 12 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml:160]: The Broker Instance name here is a required name(of the Broker instance) to be filled not a default name; The name shows up only in `kubectl get deployment -n *(namespace name)` not in the NATS connection of the Meshery UI.

File: install/kubernetes/helm/meshery-operator/crds/crds.yaml (hunk 10 above)
Comment (TomiwaAribisala-git) [reply install/kubernetes/helm/meshery-operator/crds/crds.yaml]: Regarding the port, there was an oversight on my part not reading the Meshery Docs extensively on this description, I have rectified this with a new description which extends the external endpoint not the internal.

**Comment by Ashparshp** [comment]
@TomiwaAribisala-git, let's discuss this PR during the Meshery dev meeting call, if you would. Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?usp=drivesdk).

**Comment by TomiwaAribisala-git** [comment]
> @TomiwaAribisala-git, let's discuss this PR during the Meshery dev meeting call, if you would. Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?usp=drivesdk).

Done.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**